### PR TITLE
Revert "NATS reconnect fully recreates TCP session"

### PR DESCRIPTION
**Pull request by vkuznecovas**
This reverts commit c7fbce98ba7084698675f74a857d07d5515ff9e3.

### CODE REVIEW

**Comment by vkuznecovas** [comment]
Closes #1820

**Comment by zolia** [comment]
Looks, like this nats (disconnect fix) really brokes current broker messaging. @Waldz should we revert and wait for p2p communication implementation for direct messaging?

**Comment by anjmao** [comment]
Reopen should:
1. Recreate all active subscribes as they are still using old conn.
2. Republish all messages which was in progress during Reopen but didn't success. If not whose messages will be lost.

File: go.mod [hunk 1]
```diff
@@ -74,3 +74,5 @@ require (
 	gopkg.in/urfave/cli.v1 v1.20.0 // indirect
 	gopkg.in/yaml.v2 v2.2.8 // indirect
 )
+
+replace github.com/nats-io/go-nats v1.4.0 => github.com/mysteriumnetwork/nats.go v1.4.1-0.20200303115848-b4a5324c56ed
```
Comment (soffokl) [inline go.mod]: Is it really required to have a custom version of nats?

**Comment by soffokl** [comment]
This PR was approved as a revert of the commit. But now it merged with completely different changes without review. :/

**Pull request merged**